src/wp-includes/class-wp-block-list.php

```diff
@@ -56,7 +56,13 @@ public function __construct( $blocks, $available_context = array(), $registry =
 			$registry = WP_Block_Type_Registry::get_instance();
 		}
 
-		$this->blocks            = $blocks;
+		/**
+		 * Filters blocks passed to WP_Block_List constructor.
+		 *
+		 * @since 5.9.0
+		 * @param array[]|WP_Block[] $blocks Array of parsed block data, or block instances.
+		 */
+		$this->blocks            = apply_filters( 'block_list_blocks', $blocks );
```

Comment:
Adding a filter here will cause a performance hit as this constructor is used for more than the nav block. The original request was to gain access to the nav blocks.

Rather than adding the filter here, maybe consider adding it  directly to the nav block itself, i.e. maybe in `render_block_core_navigation()`. In doing so, the list is targeted specifically for the nav blocks rather than all block list instances.